lib/Targets/AIETargetIPU.cpp

```diff
@@ -0,0 +1,148 @@
+//===- AIETargetIPU.cpp -----------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023 Advanced Micro Devices, Inc.
+//
+//===----------------------------------------------------------------------===//
+
+#include "AIETargets.h"
+
+#include "aie/Dialect/AIEX/IR/AIEXDialect.h"
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Tools/mlir-translate/MlirTranslateMain.h"
+
+#include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/Format.h"
+
+using namespace mlir;
+using namespace xilinx;
+using namespace xilinx::AIE;
+using namespace xilinx::AIEX;
+
+static void emitProlog(llvm::raw_ostream &os) {
+  os << "00000011\n";
+  os << "01000405\n";
+  os << "01000100\n";
+  os << "0B590100\n";
+  os << "000055FF\n";
+  os << "00000001\n";
+  os << "00000010\n";
+  os << "314E5A5F\n";
+  os << "635F5F31\n";
+  os << "676E696C\n";
+  os << "39354E5F\n";
+  os << "6E693131\n";
+  os << "5F727473\n";
+  os << "64726F77\n";
+  os << "00004573\n";
+  os << "07BD9630\n";
+  os << "000055FF\n";
```

Comment:
We should have a random straw poll to ask whether this is so obvious it does not deserve a single line of comment...